demo/ts/components/theme-builder/index.tsx

```diff
@@ -0,0 +1,228 @@
+import React from "react";
+import {
+  ColorScalePropType,
+  VictoryTheme,
+  VictoryThemeDefinition,
+} from "victory-core";
+import ThemePicker from "./theme-picker";
+import { VictoryChart } from "victory-chart";
+import { VictoryAxis } from "victory-axis";
+import { VictoryStack } from "victory-stack";
+import { VictoryBar } from "victory-bar";
+import { VictoryArea } from "victory-area";
+import CustomOptions from "./custom-options";
+
+export type ThemeOption = {
+  name: string;
+  config?: VictoryThemeDefinition;
+};
+
+const themes: ThemeOption[] = [
+  { name: "Clean", config: VictoryTheme.clean },
+  { name: "Material", config: VictoryTheme.material },
+  { name: "Grayscale", config: VictoryTheme.grayscale },
+];
+
+const sampleStackData = [
+  {
+    x: 1,
+    y: 2,
+  },
+  {
+    x: 2,
+    y: 3,
+  },
+  {
+    x: 3,
+    y: 5,
+  },
+  {
+    x: 4,
+    y: 4,
+  },
+  {
+    x: 5,
+    y: 7,
+  },
+];
+
+const containerStyles: React.CSSProperties = {
+  display: "flex",
+  flexDirection: "row",
+  flexWrap: "wrap",
+  alignItems: "flex-start",
+  justifyContent: "flex-start",
+  width: "100%",
+};
+
+const sidebarStyles: React.CSSProperties = {
+  height: "100%",
+  borderRight: "1px solid #ccc",
+  padding: "20px",
+  width: 300,
+};
+
+const tabContainerStyles: React.CSSProperties = {
+  display: "flex",
+  cursor: "pointer",
+  marginBottom: 20,
+  fontSize: 14,
+};
+
+const previewContainerStyles: React.CSSProperties = {
+  flex: 1,
+  padding: "0 20px",
+};
+
+const getTabStyles = (isActive): React.CSSProperties => ({
+  padding: "10px 20px",
+  borderBottom: "2px solid lightgray",
+  fontWeight: "bold",
+  color: "gray",
+  ...(isActive && {
+    borderBottom: "2px solid black",
+    color: "#2165E3",
+  }),
+});
+
+const chartsGridStyles: React.CSSProperties = {
+  display: "grid",
+  gridTemplateColumns: "1fr 1fr",
+  gap: 20,
+};
+
+const chartStyle: { [key: string]: React.CSSProperties } = {
+  parent: {
+    border: "1px solid #ccc",
+    width: "100%",
+    height: 400,
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+};
+
+const tabConfig = [
+  {
+    name: "themes",
+    label: "Default Themes",
+    Children: ThemePicker,
+  },
+  {
+    name: "customize",
+    label: "Customize",
+    Children: CustomOptions,
+  },
+];
+
+const ThemeBuilder = () => {
+  const [activeTheme, setActiveTheme] = React.useState<ThemeOption>(themes[0]);
+  const [activeColorScale] = React.useState<ColorScalePropType | undefined>(
```

Comment:
can this state be removed? doesn't look like it's being changed anywhere

Reply:
I'm going to add a handler for adjusting the color scale in my next PR so I added this for now so that the charts could read from this state.